libs/auth/src/openssl.ts

```diff
@@ -186,68 +211,254 @@ export async function verifySignature({
   const cert = await openssl([
     'x509',
     '-req',
     '-CA',
-    signingCertAuthorityCert,
-    '-CAkey',
-    signingPrivateKey,
-    '-passin',
-    `pass:${signingPrivateKeyPassword}`,
+    signingCertAuthorityCertPath,
+    ...certAuthorityKeyParams,
     '-CAcreateserial',
     '-CAserial',
     '/tmp/serial.txt',
     '-in',
     certSigningRequest,
-    '-force_pubkey',
-    publicKeyToSign,
     '-days',
     `${expiryInDays}`,
-    ...(certType === 'certAuthorityCert'
+    ...(certPublicKeyOverride ? ['-force_pubkey', certPublicKeyOverride] : []),
+    ...(certType === 'cert_authority_cert'
       ? ['-extensions', 'v3_ca', '-extfile', OPENSSL_CONFIG_FILE_PATH]
       : []),
   ]);
   return cert;
 }
+
+/**
+ * The input to createCert. All file keys and inline keys should be in PEM format.
+ */
+export interface CreateCertInput {
+  /**
+   * A private key should be provided when possible, but if the private key of the public key to
+   * sign isn't available (as is the case for a key pair generated on a Java Card), the public key is
+   * also acceptable. We set the cert key using -force_pubkey in this case.
+   */
+  certKeyInput:
+    | { type: 'private'; key: FileKey }
+    | { type: 'public'; key: InlineKey };
+  certSubject: string;
+  certType?: CertType;
+  expiryInDays: number;
+  signingCertAuthorityCertPath: string;
+  signingPrivateKey: FileKey | TpmKey;
+}
+
+const CreateCertInputSchema: z.ZodSchema<CreateCertInput> = z.object({
+  certKeyInput: z.union([
+    z.object({ type: z.literal('private'), key: FileKeySchema }),
+    z.object({ type: z.literal('public'), key: InlineKeySchema }),
+  ]),
+  certSubject: z.string(),
+  certType: z
+    .union([z.literal('cert_authority_cert'), z.literal('standard_cert')])
+    .optional(),
+  expiryInDays: z.number(),
+  signingCertAuthorityCertPath: z.string(),
+  signingPrivateKey: z.union([FileKeySchema, TpmKeySchema]),
+});
+
+/**
+ * Parses a JSON string as a CreateCertInput, throwing an error if parsing fails
+ */
+export function parseCreateCertInput(value: string): CreateCertInput {
+  return unsafeParse(CreateCertInputSchema, JSON.parse(value));
+}
+
+/**
+ * A helper for createCert. Creates a cert, combining cert signing request creation and cert
+ * creation into a single function. The outputted cert will be in PEM format.
+ */
+export async function createCertHelper({
+  certKeyInput,
+  certSubject,
+  certType = 'standard_cert',
+  expiryInDays,
+  signingCertAuthorityCertPath,
+  signingPrivateKey,
+}: CreateCertInput): Promise<Buffer> {
+  const isPrivateKeyOfPublicKeyToSignUnavailable =
+    certKeyInput.type === 'public';
+
+  let certKey: FileKey | InlineKey;
+  if (isPrivateKeyOfPublicKeyToSignUnavailable) {
+    const throwawayPrivateKey = await openssl([
+      'ecparam',
+      '-genkey',
+      '-name',
+      'prime256v1',
+      '-noout',
+    ]);
+    certKey = {
+      source: 'inline',
+      content: throwawayPrivateKey.toString('utf-8'),
+    };
```

Comment:
this happens to be fairly fast because we're using elliptic curves, but other algorithms, generation is not so fast. So if this is a throwaway key that's not actually used, I would use a pre-generated dummy one every time.

Reply:
what I mean by not-so-fast: RSA could take a few *seconds*.

Reply:
Good call! https://github.com/votingworks/vxsuite/pull/3433/commits/f66ebda79ebe995cbf9aaeb054fd7dc063a2b78e